lib/browser/platform/ios.rb

```diff
@@ -3,10 +3,21 @@ module Browser
   class Platform
     class IOS < Base
       MATCHER = /(iPhone|iPad|iPod)/
-      VERSION_MATCHER = /OS ([\d.]+)/
+      VERSION_MATCHER = /OS ((?<major>\d)_(?<minor>\d)_?(?<patch>\d)?)/
 
       def version
-        ua[VERSION_MATCHER, 1] || "0"
+        matches = VERSION_MATCHER.match(ua)
```

Comment:
I don't think we should care about backwards compatibility here. One shouldn't check strict versions with this gem. You should either make strings comparison (e.g. `browser.version >= "9.0"`) or use `browser.ios?` method. Makes sense?